mlflow/R/mlflow/R/tracking-utils.R

```diff
@@ -29,6 +29,12 @@ mlflow_get_or_start_run <- function() {
   mlflow_active_run() %||% mlflow_start_run()
 }
 
+mlflow_infer_experiment_id <- function(experiment_id) {
```

Comment:
Do we also want to have an analogous function for pulling experiment name from environment variables (e.g. to honor `MLFLOW_EXPERIMENT_NAME` in R)? Could do it in a follow-up

Reply:
Good catch. I think the MLFLOW_EXPERIMENT_NAME should be checked in the same function. Basically we should take what user gave us or make best effort to infer it. It's easier if we just infer id here.